pkg/controller/core/workload_controller.go

```diff
@@ -194,6 +194,25 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	maxTime := workload.GetMaxRetryTime(&wl)
+	if maxTime.After(r.clock.Now()) {
+		var updated bool
+		err := workload.PatchAdmissionStatus(ctx, r.client, &wl, r.clock, func() (*kueue.Workload, bool, error) {
+			if wl.Status.RequeueState == nil {
+				wl.Status.RequeueState = &kueue.RequeueState{}
+			}
+			log.V(2).Info("At least one admission check set a retry time", "maxTime", maxTime, "requeueState.RequeueAt", wl.Status.RequeueState.RequeueAt)
+			if updatedRequeueAt := workload.SetRequeueState(&wl, maxTime, false); !updatedRequeueAt {
```

Comment:
To improve readability, would it be feasible to move out the check if there is any update needed outside the PatchAdmissionStatus helper, I think even up to `if maxTime.After(r.clock.Now()) {`, replace  with more generic 
```golang
if requeueAt := workload.NeedsRequeueAtUpdate(&wl); requeueAt != nil {
```
then we don't need to track with `updated`

Reply:
Yeah, that makes sense! Let me rewrite this part.